### PR TITLE
Binderator improvements metadata files and dependency reporting

### PR DESCRIPTION
**Pull request by moljac**
Added 

* metadata downloads

* javadoc downloads

  * MavenNet javadoc support
  
    * PR https://github.com/Redth/MavenNet/pull/6

* formatted output for missing artifacts

  * summary (number of exceptions - missing artifacts)
  
  * details about missing artifacts with suggestion code snippet to be added to `config.json`
  
    ```
    Dependency errors : 1
    1
	    System.Exception: 
    No matching artifact config found for: 
			    com.google.firebase.firebase-appcheck-interop:16.0.0-beta01
    to satisfy dependency of: 
			    com.google.firebase.firebase-database:20.0.0
    
	    Please add following json snippet to config.json:
    
    
          {
            "groupId": "com.google.firebase",
            "artifactId": "firebase-appcheck-interop",
            "version": "16.0.0-beta01",
            "nugetVersion": "CHECK THIS",
            "nugetId": "CHECK THIS",
            "dependencyOnly": true/false
          }
    ```

### CODE REVIEW

**Comment by moljac** [comment]
NOTE: stable artifact depending on prerelease

**Comment by moljac** [comment]
NOTE: MavenNet 2.3.0 not published yet

File: Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs [hunk 1]
```diff
@@ -241,11 +241,13 @@ static async Task DownloadArtifacts(MavenRepository maven, BindingConfig config,
 					}
 				}
 
-
 				if (config.DownloadJavaDocJars)
 				{
 					try
 					{
+						// using (var astrm = await maven.OpenArtifactDocsFile(mavenArtifact.GroupId, mavenArtifact.ArtifactId, version))
```
Comment (jpobst) [inline Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs:248]: Is this intentionally commented out?

**Review by jpobst — APPROVED** [review]

File: Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs (hunk 1 above)
Comment (moljac) [reply Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs:248]: yes. Not sure why Redth could not publish MavenNet. He bumped everything after my PR, but nuget is not published yet.
It is not critical, I'll add it in next round. Soon

**Pull request merged**